cache/refs.go

```diff
@@ -594,8 +596,14 @@ func (sr *immutableRef) ociDesc(ctx context.Context, dhs DescHandlers) (ocispecs
 	desc := ocispecs.Descriptor{
 		Digest:      sr.getBlob(),
 		Size:        sr.getBlobSize(),
-		MediaType:   sr.getMediaType(),
 		Annotations: make(map[string]string),
+		MediaType:   sr.getMediaType(),
+	}
+
+	if convertNonDist {
+		desc.MediaType = convert.LayerToDistributable(desc.MediaType)
+	} else {
+		desc.URLs = sr.getURLs()
```

Comment:
Wouldn't you want to check first if there are URLs and only if there are then use the non-dist mediatype. Also if there are URLs you should always use non-dist mediatype or discard the URLs.

Reply:
Good point, this fixes issues I was thinking about with having different media types for the same blob.
Fixed.